File: Python/42-和为S的两个数字.py
# -*- coding:utf-8 -*-
class Solution:
    def FindNumbersWithSum(self, array, tsum):
        # write code here
        i, j = 0, len(array) - 1
        while i <= j:
            t = array[i] + array[j]
            if t == tsum:
                return [array[i], array[j]]
            elif t < tsum:
                i += 1
            else:
                j -= 1
        return []

File: C++/42-和为S的两个数字.cpp
class Solution
{
public:
    vector<int> FindNumbersWithSum(vector<int> array, int sum)
    {
        vector<int> res;
        int i = 0, j = array.size() - 1;
        while (i <= j)
        {
            int t = array[i] + array[j];
            if (t == sum)
            {
                res.push_back(array[i]);
                res.push_back(array[j]);
                return res;
            }
            else if (t < sum)
                i++;
            else
                j--;
        }
        return res;
    }
};